lms/static/js/learner_dashboard/models/program_subscription_model.js

```diff
@@ -0,0 +1,81 @@
+import Backbone from 'backbone';
+import moment from 'moment';
+
+import DateUtils from 'edx-ui-toolkit/js/utils/date-utils';
+
+/**
+ * Model for Program Subscription Data.
+ */
+class ProgramSubscriptionModel extends Backbone.Model {
+    constructor([data = {}], urls, userPreferences, ...args) {
+        const subscriptionState = data.subscription_state?.toLowerCase();
+        const subscriptionPrice =
+            data.currency === 'USD'
+                ? `$${data.price}`
+                : `${data.price} ${data.currency}`;
+        const subscriptionUrl =
+            subscriptionState === 'active'
+                ? urls.manage_subscription_url
+                : urls.buy_subscription_url;
+
+        const hasActiveTrial =
+            subscriptionState === 'active' && data.trial_end
```

Comment:
The subscription `trial_end` is being set on the subs `/stripe-subscribe` endpoint when a learner clicks on "Subscribe", so at this point, you won't have this date from the `/stripe-subscription` endpoint.

Not sure if this should be hard coded in here, or if it should be hardcoded in the endpoint.